uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/LayoutRESTController.java

```diff
@@ -115,65 +112,31 @@ public ModelAndView getRESTController(
-                for (int i = 0; i < folders.getLength(); i++) {
-                    Node node = folders.item(i);
-                    if (tab.equalsIgnoreCase(
-                            node.getAttributes().getNamedItem("name").getNodeValue())) {
-                        TabListOfNodes tabNodes = new TabListOfNodes();
-                        tabNodes.addAllChannels(node.getChildNodes());
-                        portletNodes = tabNodes;
-                        break;
-                    }
-                }
-            } else {
-                portletNodes = document.getElementsByTagName("channel");
-            }
+            final Document document = userLayout.getLayout();
+
+            final NodeList portletNodes = getNodeList(tab, document);
+
             for (int i = 0; i < portletNodes.getLength(); i++) {
                 try {
 
-                    NamedNodeMap attributes = portletNodes.item(i).getAttributes();
-                    IPortletDefinition def =
+                    final NamedNodeMap attributes = portletNodes.item(i).getAttributes();
+                    final IPortletDefinition def =
                             portletDao.getPortletDefinitionByFname(
                                     attributes.getNamedItem("fname").getNodeValue());
-                    LayoutPortlet portlet = new LayoutPortlet(def);
+                    final LayoutPortlet portlet = new LayoutPortlet(def);
 
                     portlet.setNodeId(attributes.getNamedItem("ID").getNodeValue());
 
                     // get alt max URL
-                    String alternativeMaximizedLink = def.getAlternativeMaximizedLink();
-                    if (alternativeMaximizedLink != null) {
-                        portlet.setUrl(alternativeMaximizedLink);
-                        portlet.setAltMaxUrl(true);
-                    } else {
-                        // get the maximized URL for this portlet
-                        final IPortalUrlBuilder portalUrlBuilder =
-                                urlProvider.getPortalUrlBuilderByLayoutNode(
-                                        request,
-                                        attributes.getNamedItem("ID").getNodeValue(),
-                                        UrlType.RENDER);
-                        final IPortletWindowId targetPortletWindowId =
-                                portalUrlBuilder.getTargetPortletWindowId();
-                        if (targetPortletWindowId != null) {
-                            final IPortletUrlBuilder portletUrlBuilder =
-                                    portalUrlBuilder.getPortletUrlBuilder(targetPortletWindowId);
-                            portletUrlBuilder.setWindowState(WindowState.MAXIMIZED);
-                        }
-                        portlet.setUrl(portalUrlBuilder.getUrlString());
-                        portlet.setAltMaxUrl(false);
-                    }
+                    setAltMaxURL(request, attributes, def, portlet);
```

Comment:
It may be better to refactor the code in another commit ?

Reply:
plz see comment above